Exception in case of empty documents

## Expected Behavior
If an empty document is searched for and found it should be returned.

## Actual Behavior
In case of `validate=True` the following exception is thrown:
```
s1 = set(), s2 = set()

    def _jaccard_similarity(s1: Iterable, s2: Iterable):
        if not isinstance(s1, set):
            s1 = set(s1)
        if not isinstance(s2, set):
            s2 = set(s2)
>       return len(s1.intersection(s2)) / len(s1.union(s2))
E       ZeroDivisionError: division by zero
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 
 ## [Unreleased]
+### Fixed
+- Fixes #63 which led to Exceptions in case of empty documents.
 
 ## [0.9.2] - 2022-03-29
 ### Fixed

--- a/narrow_down/similarity_store.py
+++ b/narrow_down/similarity_store.py
@@ -336,4 +336,7 @@ def _jaccard_similarity(s1: Iterable, s2: Iterable):
         s1 = set(s1)
     if not isinstance(s2, set):
         s2 = set(s2)
-    return len(s1.intersection(s2)) / len(s1.union(s2))
+    union = s1.union(s2)
+    if not union:
+        return 1.0 if len(s1) == len(s2) == 0 else 0.0
+    return len(s1.intersection(s2)) / len(union)

--- a/tests/test_similarity_store.py
+++ b/tests/test_similarity_store.py
@@ -98,6 +98,22 @@ async def test_similarity_store__query_with_validation(monkeypatch, storage_leve
 
 
 @pytest.mark.asyncio
+async def test_similarity_store__query_empty_result_with_validation(monkeypatch):
+    async def fake_query(*args, **kwargs):
+        return [
+            StoredDocument(id_=1, document=""),
+        ]
+
+    simstore = await SimilarityStore.create(
+        storage_level=StorageLevel.Document, tokenize="char_ngrams(1)"
+    )
+    monkeypatch.setattr(simstore._lsh, "query", fake_query)
+
+    results = await simstore.query(document="", validate=True)
+    assert results == []
+
+
+@pytest.mark.asyncio
 @pytest.mark.parametrize(
     "validate",
     [True, False, None],

--- a/tests/test_similarity_store.py
+++ b/tests/test_similarity_store.py
@@ -110,7 +110,7 @@ async def test_similarity_store__query_empty_result_with_validation(monkeypatch)
     monkeypatch.setattr(simstore._lsh, "query", fake_query)
 
     results = await simstore.query(document="", validate=True)
-    assert results == []
+    assert results == [StoredDocument(id_=1, document="")]
 
 
 @pytest.mark.asyncio

